vike/shared/getPageContextRequestUrl.ts

```diff
@@ -11,6 +11,7 @@ const doNotCreateExtraDirectory = false
 
 // See node/renderPage/handlePageContextRequestUrl.ts
 function getPageContextRequestUrl(url: string): string {
-  const pageContextRequestUrl = urlToFile(url, pageContextJsonFileExtension, doNotCreateExtraDirectory)
+  let pageContextRequestUrl = urlToFile(url, pageContextJsonFileExtension, doNotCreateExtraDirectory)
+  if(url.endsWith('/')) pageContextRequestUrl += '/'
```

Comment:
I'm not sure if this is the right way to do it. Other parts depend on `getPageContextRequestUrl()` and will fail now, because requests to `/test/index.pageContext.json/` cannot be resolved. How to proceed with this?

Reply:
See the modules that import `export { pageContextJsonFileExtension }`. (Feel free to let me know if this is getting too complex; I can take over the PR.)

Reply:
If you have the patience with me to iterate a few times, until I got it right, I'm happy to work on it. So I can get to know Vike better, contribute a little bit back and learn a bunch of things.

Reply:
If it's fun and/or interesting for you, and you don't mind minimal answers from my side, then let's continue!

Reply:
Note: Adding a query param like `trailingSlash` is less intrusive, but has the downside that `trailingSlash` becomes a reserved param in Vike.

Reply:
Or maybe `/en//index.pageContext.json` after all.

Reply:
Note: Having a trailing slash `/en/index.pageContext.json/` doesn't really work out. To reach the goal the request must be made to `/en/index.pageContext.json/`, that not resolves to a file without changing the webserver routing, whereas `/en//index.pageContext.json` just works. Even though this is a quirk, because [RFC2396](www.ietf.org/rfc/rfc2396.txt) defines path separator as a *single* slash.

Reply:
> Or maybe /en//index.pageContext.json after all.

I will follow that path and then we can have a look at the implementation.

Reply:
Good to know :+1: 

My concern about `pageContext.json?trailingSlash` is that the URL doesn't end with `.json` anymore. I don't know which of the two is the lesser evil :thinking:

An alternative would be to add a property `pageContext._urlHasTrailingSlash` that is included in `pageContext.json` (i.e. in the default `passToClient` values see `PASS_TO_CLIENT`). That would be the cleanest I guess but maybe a little bit more cumbersome to implement.